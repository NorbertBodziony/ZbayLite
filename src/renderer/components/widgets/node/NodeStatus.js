import React from 'react'
import PropTypes from 'prop-types'

import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'
import PulseDot from '../../ui/PulseDot'

const styles = theme => ({
  root: {},
  status: {
    paddingLeft: theme.spacing.unit
  }
})

export const NodeStatus = ({ classes, status, percentSynced }) => (
  <Grid container>
    <Grid item>
      <PulseDot size={8} color={status} />
      <Typography inline variant='overline' className={classes.status}>
        {status} { percentSynced !== null && `${percentSynced}%`}
      </Typography>
    </Grid>
  </Grid>
)

NodeStatus.propTypes = {
  classes: PropTypes.object.isRequired,
  status: PropTypes.oneOf(['healthy', 'syncing', 'restarting', 'down', 'connecting']).isRequired,
  percentSynced: PropTypes.string
}

NodeStatus.defaultProps = {
  percentSynced: null
}

export default React.memo(withStyles(styles)(NodeStatus))
