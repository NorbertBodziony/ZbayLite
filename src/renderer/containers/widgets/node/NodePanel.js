import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import NodePanelComponent from '../../../components/widgets/node/NodePanel'
import nodeHandlers from '../../../store/handlers/node'
import { useInterval } from '../../hooks'

export const mapDispatchToProps = dispatch => bindActionCreators({
  getStatus: nodeHandlers.epics.getStatus
}, dispatch)

export const NodePanel = ({ className, getStatus }) => {
  useInterval(getStatus, 15000)
  return (
    <NodePanelComponent className={className} />
  )
}

export default connect(null, mapDispatchToProps)(NodePanel)
