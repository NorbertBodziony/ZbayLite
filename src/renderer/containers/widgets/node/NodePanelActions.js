import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import NodePanelActions from '../../../components/widgets/node/NodePanelActions'

import nodeHandlers from '../../../store/handlers/node'

export const mapDispatchToProps = dispatch => bindActionCreators({
  onRestart: nodeHandlers.epics.restart,
  onPower: nodeHandlers.epics.togglePower
}, dispatch)

export default connect(null, mapDispatchToProps)(NodePanelActions)
