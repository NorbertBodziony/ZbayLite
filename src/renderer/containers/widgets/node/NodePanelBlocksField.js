import React from 'react'
import { connect } from 'react-redux'

import Typography from '@material-ui/core/Typography'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

import nodeSelectors from '../../../store/selectors/node'

export const mapStateToProps = state => ({
  latestBlock: nodeSelectors.latestBlock(state),
  currentBlock: nodeSelectors.currentBlock(state)
})

export const NodePanelBlocksField = ({ latestBlock, currentBlock }) => (
  <NodePanelField name='Blocks'>
    <Typography inline variant='overline'>
      {currentBlock} / {latestBlock || '?'}
    </Typography>
  </NodePanelField>
)

export default connect(mapStateToProps)(NodePanelBlocksField)
