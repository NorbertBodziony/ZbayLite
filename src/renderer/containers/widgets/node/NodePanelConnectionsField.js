import React from 'react'
import BigNumber from 'bignumber.js'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import Typography from '@material-ui/core/Typography'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

import nodeSelectors from '../../../store/selectors/node'

export const mapStateToProps = state => ({
  connections: nodeSelectors.connections(state)
})

export const NodePanelConnectionsField = ({ connections }) => (
  <NodePanelField name='Connections'>
    <Typography inline variant='overline'>
      {connections.toString()}
    </Typography>
  </NodePanelField>
)

NodePanelConnectionsField.propTypes = {
  connections: PropTypes.instanceOf(BigNumber).isRequired
}

export default connect(mapStateToProps)(NodePanelConnectionsField)
