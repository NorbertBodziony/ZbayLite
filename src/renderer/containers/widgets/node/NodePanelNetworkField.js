import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import Typography from '@material-ui/core/Typography'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

import nodeSelectors from '../../../store/selectors/node'

export const mapStateToProps = state => ({
  network: nodeSelectors.network(state)
})

export const NodePanelNetworkField = ({ network }) => (
  <NodePanelField name='Network'>
    <Typography inline variant='overline'>
      {network || '?'}
    </Typography>
  </NodePanelField>
)

NodePanelNetworkField.propTypes = {
  network: PropTypes.oneOf(['mainnet', 'testnet'])
}

export default connect(mapStateToProps)(NodePanelNetworkField)
