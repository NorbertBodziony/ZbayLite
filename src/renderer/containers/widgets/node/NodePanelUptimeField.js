import React from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'
import { connect } from 'react-redux'

import Typography from '@material-ui/core/Typography'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

import nodeSelectors from '../../../store/selectors/node'

export const mapStateToProps = state => ({
  uptime: nodeSelectors.uptime(state)
})

export const NodePanelUptimeField = ({ uptime }) => (
  <NodePanelField name='Uptime'>
    <Typography display='inline' variant='overline'>
      {R.isEmpty(uptime) ? '-' : null}
      {uptime.days ? `${uptime.days}d  ` : null}
      {uptime.hours ? `${uptime.hours}h  ` : null}
      {uptime.minutes ? `${uptime.minutes}m` : null}
    </Typography>
  </NodePanelField>
)

NodePanelUptimeField.propTypes = {
  uptime: PropTypes.exact({
    days: PropTypes.number,
    hours: PropTypes.number,
    minutes: PropTypes.number,
    seconds: PropTypes.number
  }).isRequired
}

export default connect(mapStateToProps)(NodePanelUptimeField)
