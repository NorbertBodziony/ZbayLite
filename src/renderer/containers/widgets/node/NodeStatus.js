import { connect } from 'react-redux'

import NodeStatus from '../../../components/widgets/node/NodeStatus'

import nodeSelectors from '../../../store/selectors/node'
import identitySelectors from '../../../store/selectors/identity'

export const mapStateToProps = state => ({
  status: nodeSelectors.status(state),
  percentSynced: nodeSelectors.percentSynced(state),
  freeUtxos: identitySelectors.freeUtxos(state)
})

export default connect(mapStateToProps)(NodeStatus)
