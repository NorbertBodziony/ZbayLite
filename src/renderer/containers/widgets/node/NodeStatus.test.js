/* eslint import/first: 0 */
jest.mock('../../../vault')
import Immutable from 'immutable'

import { mapStateToProps } from './NodeStatus'

import create from '../../../store/create'
import { NodeState } from '../../../store/handlers/node'

describe('NodeStatus', () => {
  let store = null
  beforeEach(() => {
    jest.clearAllMocks()
    store = create({
      initialState: Immutable.Map({
        node: NodeState({
          status: 'down',
          latestBlock: 100,
          currentBlock: 18
        })
      })
    })
  })

  it('will receive right props', async () => {
    const props = mapStateToProps(store.getState())
    expect(props).toMatchSnapshot()
  })
})
