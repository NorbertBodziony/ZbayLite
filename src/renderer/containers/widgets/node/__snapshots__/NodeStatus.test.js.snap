// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeStatus will receive right props 1`] = `
Object {
  "connections": 0,
  "freeUtxos": 0,
  "percentSynced": "18",
  "status": "down",
}
`;
